Frontend/src/Components/Faults/FaultsView.js

```diff
@@ -1,13 +1,334 @@
+        return (props.data?.motor_temp[0] > 100) || (props.data?.driverIO_temp[0] > 100) ||
+               (props.data?.mainIO_temp[0] > 100) || (props.data?.cabin_temp[0] > 100) ||
+               (props.data?.string1_temp[0] > 50) || (props.data?.string1_temp[0] < 20) ||
+               (props.data?.string2_temp[0] > 50) || (props.data?.string2_temp[0] < 20) ||
+               (props.data?.string3_temp[0] > 50) || (props.data?.string3_temp[0] < 20) ||
+               (props.data?.pack_temp[0] > 60);
+    }
+
+    const getTempString = () => {
+        let highTempStrings = [];
+        let lowTempStrings = [];
+
+        if(props.data?.motor_temp[0] > 100) {
+            highTempStrings.push("Motor");
+        }
+        if(props.data?.driverIO_temp[0] > 100) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Driver IO" : "driver IO");
+        }
+        if(props.data?.mainIO_temp[0] > 100) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Main IO" : "main IO");
+        }
+        if(props.data?.cabin_temp[0] > 100) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Cabin" : "cabin");
+        }
+        if(props.data?.string1_temp[0] > 50) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Cell string 1" : "cell string 1");
+        } else if(props.data?.string1_temp[0] < 20) {
+            lowTempStrings.push("Cell string 1");
+        }
+        if(props.data?.string2_temp[0] > 50) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Cell string 2" : "cell string 2");
+        } else if(props.data?.string2_temp[0] < 20) {
+            lowTempStrings.push((lowTempStrings.length == 0) ? "Cell string 2" : "cell string 2");
+        }
+        if(props.data?.string3_temp[0] > 50) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Cell string 3" : "cell string 3");
+        } else if(props.data?.string3_temp[0] < 20) {
+            lowTempStrings.push((lowTempStrings.length == 0) ? "Cell string 3" : "cell string 3");
+        }
+        if(props.data?.pack_temp[0] > 60) {
+            highTempStrings.push((highTempStrings.length == 0) ? "Battery pack" : "battery pack");
+        }
+
+        let highTempString = "";
+        let lowTempString = "";
+
+        if(highTempStrings.length > 0) {
+            if(highTempStrings.length == 1) {
+                highTempString = highTempStrings[0] + " temp is high";
+            } else if(highTempStrings.length == 2) {
+                highTempString = highTempStrings[0] + " and " + highTempStrings[1] + " temps are high";
+            } else {
+                highTempString = highTempStrings[0];
+                for(let i=1; i < highTempStrings.length - 1; i++) {
+                    highTempString += ", " + highTempStrings[i];
+                }
+                highTempString += ", and " + highTempStrings[highTempStrings.length - 1] + " temps are high";
+            }
+        }
```

Comment:
```suggestion
          if(highTempStrings.length == 1) {
              highTempString = highTempStrings[0] + " temp is high";
          } else if(highTempStrings.length == 2) {
              highTempString = highTempStrings[0] + " and " + highTempStrings[1] + " temps are high";
          } else {
              highTempString = highTempStrings[0];
              for(let i=1; i < highTempStrings.length - 1; i++) {
                    highTempString += ", " + highTempStrings[i];
              }
              highTempString += ", and " + highTempStrings[highTempStrings.length - 1] + " temps are high";
          }
```
Formatting is probably bad but you get the point. If `length == 1` then `length > 0` so no need for 2 ifs